src/Database/LSMTree/Internal/WriteBufferBlobs.hs

```diff
@@ -169,22 +189,47 @@ mkWeakBlobRef (DeRef WriteBufferBlobs {blobFile}) blobspan =
       weakBlobRefSpan = blobspan
     }
 
-
 -- | A mutable file offset, suitable to share between threads.
+#if MIN_VERSION_base(4,16,1)
+newtype FilePointer m = FilePointer (PrimVar (PrimState m) Word)
+#else
 newtype FilePointer m = FilePointer (PrimVar (PrimState m) Int)
---TODO: this would be better as Word64
--- this will limit to 31bit file sizes on 32bit arches
```

Comment:
I wonder: is it even possible to do atomic operations on `Word64` on 32 bit architectures? Doesn't a 32-bit architecture need to represent a `64-bit` word as two machine words, losing the ability to modify both at the same time atomically?

Reply:
While it is probably chip dependant, I would imagine the answer is generally "yes," a 32-bit machine cannot be guaranteed to operate on `Word64` in an atomic manner. Is this a deal breaker for us?

Reply:
If we really want to have 64-bit file pointers on 32-bit systems, then we'd have to use something like `MVar` on those systems. Seeing as we don't support 32-bit systems, I think we don't have to consider that case